python/paddle/tensor/math.py

```diff
@@ -5657,7 +5657,7 @@ def _gcd_body_fn(x, y):
         )
         return (paddle.where(x < y, y, x), paddle.where(x < y, x, y))
 
-    if in_dynamic_mode():
+    if in_dynamic_or_pir_mode():
```

Comment:
因为 pir 下与控制流相关的算子适配了静态图专用 api (比如下面的 `paddle.static.nn.while_loop` )，所以涉及控制流 api 可以与原有的旧静态图共用分支。这里可以复原
```suggestion
    if in_dynamic_mode():
```

Reply:
收到